quinn-proto/src/connection/streams/recv.rs

```diff
@@ -248,6 +248,27 @@ pub struct DidRead<T> {
     pub(crate) max_data: ShouldTransmit,
 }
 
+impl<T> DidRead<T> {
+    /// Returns `true` iff flow control credit should be issued.
+    pub fn should_issue_flow_control_credit(&self) -> bool {
+        self.max_stream_data.should_transmit() || self.max_data.should_transmit()
+    }
+
+    /// Maps an `DidRead<T>` to `DidRead<U>` by applying a function to the contained result.
+    pub fn map<U, F: FnOnce(T) -> U>(self, f: F) -> DidRead<U> {
+        DidRead {
+            result: f(self.result),
+            max_stream_data: self.max_stream_data,
+            max_data: self.max_data,
+        }
+    }
+
+    /// Returns the contained result, consuming the `self` value.
+    pub fn unwrap(self) -> T {
```

Comment:
`unwrap` is conventionally used for functions that may panic; let's call this `into_inner` or `into_result` or similar instead.

Reply:
OK, I renamed it to `into_inner`.